src/rust/src/padding.rs

```diff
@@ -169,6 +224,67 @@ impl PKCS7UnpaddingContext {
     }
 }
 
+#[pyo3::pyclass]
+pub(crate) struct ANSIX923UnpaddingContext {
+    block_size: usize,
+    buffer: Option<Vec<u8>>,
+}
+
+#[pyo3::pymethods]
+impl ANSIX923UnpaddingContext {
+    #[new]
+    pub(crate) fn new(block_size: usize) -> ANSIX923UnpaddingContext {
+        ANSIX923UnpaddingContext {
+            block_size: block_size / 8,
+            buffer: Some(Vec::new()),
+        }
+    }
+
+    pub(crate) fn update<'p>(
+        &mut self,
+        buf: CffiBuf<'p>,
+        py: pyo3::Python<'p>,
+    ) -> CryptographyResult<pyo3::Bound<'p, pyo3::types::PyBytes>> {
+        match self.buffer.as_mut() {
+            Some(v) => {
+                v.extend_from_slice(buf.as_bytes());
+                let finished_blocks =
+                    std::cmp::max((v.len() / self.block_size).saturating_sub(1), 0);
```

Comment:
Is a negative value possible here? If not, why do we have the `max()` call? (If it is possible, we have the same bug in pkcs7unpadder I guess)

Reply:
This is a direct port of this line in the Python implementation:

https://github.com/pyca/cryptography/blob/46a8693ad634d7a2a606a4cbb0ee759147dd911c/src/cryptography/hazmat/primitives/padding.py#L81

I'll think about whether or not the `max` was ever really necessary.

Reply:
python doesn't have saturating sub :-)